File: backend/app/schemas/contract.py
# backend/app/schemas/contract.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ContractResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    deal_id: uuid.UUID
    public_id: str
    status: str
    sent_at: datetime | None
    signed_at: datetime | None
    activated_at: datetime | None
    total_commitment_cents: int | None
    created_at: datetime


class AssetResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    contract_id: uuid.UUID
    name: str
    category: str | None
    quantity: int
    unit_value_cents: int | None
    created_at: datetime


class PaymentScheduleEntryResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    contract_id: uuid.UUID
    due_date: datetime
    amount_cents: int
    status: str
    created_at: datetime


class ActivationChecklistItem(BaseModel):
    key: str
    label: str
    satisfied: bool


class ActivationChecklistResponse(BaseModel):
    items: list[ActivationChecklistItem]
    all_satisfied: bool

File: web/lib/types/contract.ts
// web/lib/types/contract.ts
export interface Contract {
  id: string
  deal_id: string
  public_id: string
  status: string
  sent_at: string | null
  signed_at: string | null
  activated_at: string | null
  total_commitment_cents: number | null
  created_at: string
}

export interface Asset {
  id: string
  contract_id: string
  name: string
  category: string | null
  quantity: number
  unit_value_cents: number | null
  created_at: string
}

export interface PaymentScheduleEntry {
  id: string
  contract_id: string
  due_date: string
  amount_cents: number
  status: string
  created_at: string
}

export interface ActivationChecklistItem {
  key: string
  label: string
  satisfied: boolean
}

export interface ActivationChecklist {
  items: ActivationChecklistItem[]
  all_satisfied: boolean
}
